ios/browser/api/history/brave_browsing_history_driver_ios.mm

```diff
@@ -0,0 +1,92 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/browser/api/history/brave_browsing_history_driver_ios.h"
+#include <utility>
+
+#include "base/check.h"
+#include "base/strings/utf_string_conversions.h"
+
+#include "components/browsing_data/core/history_notice_utils.h"
+
+#include "ios/chrome/browser/browser_state/chrome_browser_state.h"
+#include "ios/chrome/browser/history/history_utils.h"
+#include "ios/chrome/browser/history/web_history_service_factory.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+using history::BrowsingHistoryService;
+
+#pragma mark - BraveBrowsingHistoryDriverIOS
+
+BraveBrowsingHistoryDriverIOS::BraveBrowsingHistoryDriverIOS(
+    ChromeBrowserState* browser_state,
+    id<BraveHistoryDriverDelegate> delegate)
```

Comment:
we should just be using our own implementation of `HistoryConsumer` here so we don't have to duplicate `BrowsingHistoryDriverIOS`

Reply:
Same with https://github.com/brave/brave-core/pull/8869#discussion_r646984276